kong/init.lua

```diff
@@ -293,12 +293,20 @@ function Kong.balancer()
     local previous_try = tries[addr.try_count - 1]
     previous_try.state, previous_try.code = get_last_failure()
 
-    local ok, err = balancer_execute(addr)
+    -- Report HTTP status for health checks
+    if addr.balancer then
+      if previous_try.state == "failed" then
+        addr.balancer.report_tcp_failure(addr.ip, addr.port)
+      else
+        addr.balancer.report_http_status(addr.ip, addr.port, previous_try.code)
+      end
+    end
+
+    local ok, err, errcode = balancer_execute(addr)
     if not ok then
       ngx_log(ngx_ERR, "failed to retry the dns/balancer resolver for ",
               tostring(addr.host), "' with: ", tostring(err))
-
-      return responses.send(500)
+      return ngx.exit(errcode)
```

Comment:
curious about this change away from `responses.send`?

Reply:
For 500 in particular it would work, but it's for consistency with the change above. Our `responses` module sometimes does ngx.say which causes a crash in the balancer context.